lib/browser/api/web-contents.js

```diff
@@ -206,60 +207,135 @@ WebContents.prototype.executeJavaScriptInIsolatedWorld = async function (code, h
 
 // Translate the options of printToPDF.
 WebContents.prototype.printToPDF = function (options) {
-  const printingSetting = {
+  const printSettings = {
     ...defaultPrintingSetting,
     requestID: getNextId()
   }
+
   if (options.landscape) {
-    printingSetting.landscape = options.landscape
-  }
-  if (options.fitToPageEnabled) {
-    printingSetting.fitToPageEnabled = options.fitToPageEnabled
+    if (typeof options.landscape !== 'boolean') {
```

Comment:
@ckerr in this case, does the above solution remove the need for the user to check whether there is a default value? in this case, as a new reader to this code i think this would actually fool me into thinking that there _were_ no defaults, and that i have to set one if i want it to work at all. It's also arguablt a bit less idiomatic to directly check for 'undefined' imo, at least in typical web app architecture, and instead to proceed on truthiness of whatever parameter is being passed in. However, i don't feel _incredibly_ strongly about this and would still be open to changing it!

Reply:
So upfront, I don't feel strongly about this either -- both versions of the code will produce the same output, I think. :smile_cat: 

That said, to answer the questions above...

> does the above solution remove the need for the user to check whether there is a default value?

Yes, iff the user provides a valid value. IMO that's a good thing, since a valid user-provided value should make the default moot.

> i think this may actually fool me into thinking that there were no defaults

The `const printSettings = { ...defaultPrintSettings` code at the top of this block should prevent that misconception 

> It's also arguably a bit less idiomatic to directly check for 'undefined' imo,

I don't think it's idiomatic to test whether a boolean is initialized or not by checking its truthiness?

Reply:
that's fair! i'll update this this morning :)

Reply:
resolved :D